### PR TITLE
Refactor Linux autostart

### PR DESCRIPTION
**Pull request by samuel-w**
Changed it so it reads the desktop file in src/assets/metadata instead of hardcoding it. It also uses XDG_CONFIG_HOME for non-flatpak installs (Can't do in flatpak since it gets overridden). 
Also added a test, it might break in the future since clicking qcheckboxes is weird.

### CODE REVIEW

File: src/vorta/autostart.py [hunk 1]
```diff
@@ -42,20 +33,33 @@ def open_app_at_startup(enabled=True):
...
 
-        autostart_file_path = autostart_path / 'vorta.desktop'
-
-        if enabled:
-            if Path('/.flatpak-info').exists():
-                # Vorta runs as flatpak
-                autostart_file_path.write_text(LINUX_STARTUP_FILE.format('flatpak run com.borgbase.Vorta'))
+            # Find XDG_CONFIG_HOME unless when running in flatpak
+            if is_flatpak:
+                autostart_path = Path.home() / '.config' / 'autostart'
```
[the remark this answers is not in the document]
Comment (samuel-w) [reply src/vorta/autostart.py:44]: Yeah, flatpak sets XDG_CONFIG_HOME to `$HOME/.var/app/com.borgbase.Vorta/config`.

**Comment by m3nu** [comment]
You're happy with this @Hofer-Julian ?

**Comment by Hofer-Julian** [comment]
Didn't have time to test it yet.
The changes look straightforward though, and if @samuel-w confirms that he tested it with pip and flatpak, I think it can be merged.

**Comment by samuel-w** [comment]
This was tested with pip and flatpak, just for flatpak it ignores XDG_CONFIG_HOME and installs to ~/.config/autostart. Should not be an issue, since that was the behaviour before and no one complained.

**Comment by Hofer-Julian** [comment]
Fine as soon as flatpak supports autostart natively this will be refactored anyways

**Pull request merged**